src/hip_module.cpp

```diff
@@ -230,7 +229,11 @@ hipError_t ihipModuleLaunchKernel(hipFunction_t f, uint32_t globalWorkSizeX,
 
         hc::completion_future cf;
 
-        lp.av->dispatch_hsa_kernel(&aql, config[1] /* kernarg*/, kernArgSize,
+        // Insert 48-bytes at the end for implicit kernel arguments and fill with value zero.
```

Comment:
We should move these three lines earlier, closer to where the kernargs vector is being populated. The code just above this line is not related to kernargs, and it is cleaner to have all kernarg related bunched together.

Reply:
It is done.